pokedex/storage.lua

```diff
@@ -61,49 +98,41 @@ function M.clamp_max_active_pokemon(new_max)
 	return math.max(range_min, math.max(range_min, new_max))
 end
 
+
 function M.is_party_full()
 	local counter = 0
-	for _, _ in pairs(active) do
-		counter = counter + 1
+	for _, pkmn in pairs(player_pokemon) do
```

Comment:
This is an example where you need a separate set of tables by location for things to be fast. Looping through EVERY pokemon in the PC just to figure out if your party is full is bad!